src/opnsense/mvc/app/models/OPNsense/Backup/NextcloudSettings.xml

```diff
@@ -9,6 +9,8 @@
         </enabled>
         <url type="TextField">
             <Required>N</Required>
+            <mask>/^https?:\/\/.*[^\/]$/</mask>
```

Comment:
Incidentally, I use cloudnow. It is an Austrian company. and I post here only my comment because it would be another alternative and it is synonymous of me for months running smoothly. maybe this would be another alternative?

![grafik](https://user-images.githubusercontent.com/34602360/55169387-00536680-5175-11e9-9b09-79ee09642bde.png)

Reply:
https://www.cloudfiles.app/

Reply:
https://www.cloudfiles.app/sicherheit/

Reply:
@opnsenseuser What do you want to tell me with your screenshot? It is very unlikely that the validation fails with your URL.

Reply:
sorry, I have mispronounced myself. I just wanted to say that cloudnow settings work here as well.

Reply:
I'm sure that other cloud services would also be configurable via the input mask. So would not it be advantageous to replace the term nextcloud with a generic term?

Reply:
@fabianfrz I just wanted to make an improvement proposal. I imagine a drop down box, where you can select different cloud services. eg nextcloud, owncloud, cloudnow, etc. the data are then eg already pre-filled but still modifiable. I imagine that as in the dyndns settings Plugin from opnsense. what do you mean?

Reply:
@opnsenseuser nextcloud and owncloud are usually on premise solutions, so you have to explicitly configure the URL or the hostname anyway. In my case I have my own nextcloud on my own domain which I used for testing this module against. The reason why it is called nextcloud is that I do not test it against an owncloud instance (even if it very likely works because the same WebDAV library is working under the hood).

SAAS services are different. If it works, that's fine but I don't want to test against all of them (never ending story) and I currently cannot add anything else then the input fields (for example no JS to create something like a template directly in my file but that should be pluggable). Please note that this must follow the paths of nextcloud to work. Other WebDAV variants will not work as the nextcloud paths and responses are hardcoded in the file.